### PR TITLE
vendor: go.etcd.io/etcd v3.5.16, go.etcd.io/etcd/server/v3 v3.5.16

### PR DESCRIPTION
**Pull request by thaJeztah**
### vendor: go.etcd.io/etcd v3.5.16

This dependency comes from SwarmKit, but didn't update dependencies
for quite some time. Update the dependencies to their latest patch
release.

Keeping the server at v3.5.6, because newer versions pull in some
odd dependencies (but not used in code).

changelog: https://github.com/etcd-io/etcd/blob/f0187c38d1828a3241984c976a3cd2fd4749afc6/CHANGELOG/CHANGELOG-3.5.md

full diff: https://github.com/etcd-io/etcd/compare/v3.5.6...v3.5.16

### vendor: go.etcd.io/etcd/server/v3 v3.5.16

This dependency comes from SwarmKit, but didn't update dependencies
for quite some time. Update the dependencies to their latest patch
release.

For some reason, updating this dependency brings back various legacy
versions of dependencies through `github.com/coreos/etcd/clientv3/concurrency`;

    go: finding module for package github.com/coreos/etcd/clientv3/concurrency
    go: finding module for package github.com/coreos/etcd/clientv3
    go: found github.com/coreos/etcd/clientv3 in github.com/coreos/etcd v3.3.27+incompatible
    go: found github.com/coreos/etcd/clientv3/concurrency in github.com/coreos/etcd v3.3.27+incompatible
    go: finding module for package github.com/coreos/go-systemd/journal
    go: finding module for package github.com/coreos/pkg/capnslog
    go: found github.com/coreos/pkg/capnslog in github.com/coreos/pkg v0.0.0-20240122114842-bbd7aa9bf6fb
    go: found github.com/coreos/go-systemd/journal in github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf

The above resolution results in these legacy versions to be added;

    github.com/coreos/etcd v3.3.27+incompatible // indirect
    github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf // indirect
    github.com/coreos/pkg v0.0.0-20240122114842-bbd7aa9bf6fb // indirect

However, none of those are _used_, and they don't appear as vendored file.

That change happens with v3.5.7 and up, but nothing immediately stands
out what change causes this. Possibly it's due to the switch to go1.17
in go.mod, which is the first version that flattened indirect dependencies,
potentially causing intermediate dependency versions to be used during
resolution; https://github.com/etcd-io/etcd/compare/server/v3.5.6...server/v3.5.7

changelog: https://github.com/etcd-io/etcd/blob/f0187c38d1828a3241984c976a3cd2fd4749afc6/CHANGELOG/CHANGELOG-3.5.md

full diff: https://github.com/etcd-io/etcd/compare/server/v3.5.6...server/v3.5.16

### CODE REVIEW

**Comment by thaJeztah** [comment]
Oh! I think I know; it's that rexray thing again; we don't use a replace here, but swarmkit does; https://github.com/moby/swarmkit/blob/ea1a7cec35cb05dce3cff2de3f4ee28cbca6ca79/go.mod#L94-L95

```
// Removes etcd dependency
replace github.com/rexray/gocsi => github.com/dperny/gocsi v1.2.3-pre
```

First time that `go mod why` gives me a useful result :joy:

```
go mod why github.com/coreos/etcd/clientv3
go: downloading github.com/jonboulle/clockwork v0.3.0
# github.com/coreos/etcd/clientv3
github.com/docker/docker/daemon/cluster
github.com/moby/swarmkit/v2/node
github.com/moby/swarmkit/v2/manager
github.com/moby/swarmkit/v2/manager/csi
github.com/moby/swarmkit/v2/manager/csi.test
github.com/rexray/gocsi/mock/provider
github.com/rexray/gocsi
github.com/rexray/gocsi/middleware/serialvolume/etcd
github.com/coreos/etcd/clientv3
```

I opened a PR to remove that dependency as it wasn't even used;

- https://github.com/moby/swarmkit/pull/3184

**Review by vvoland — COMMENTED** [review]
LGTM (once the swarmkit PR is merged)

**Review by vvoland — APPROVED** [review]

**Pull request merged**
